lib/matplotlib/testing/conversion_cache.py

```diff
@@ -0,0 +1,200 @@
+"""
+A cache of png files keyed by the MD5 hashes of corresponding svg and
+pdf files, to reduce test suite running times for svg and pdf files
+that stay exactly the same from one run to the next.
+
+There is a corresponding nose plugin in testing/nose/plugins and
+similar pytest code in conftest.py.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import hashlib
+import shutil
+import os
+import warnings
+
+from matplotlib import _get_cachedir
+from matplotlib import cbook
+from matplotlib import checkdep_ghostscript
+from matplotlib import checkdep_inkscape
+
+
+class ConversionCache(object):
+    """A cache that stores png files converted from svg or pdf formats.
+
+    The image comparison test cases compare svg and pdf files by
+    converting them to png files. When a test case has produced a
+    file, e.g. result.pdf, it queries this cache by the pathname
+    '/path/to/result_images/result.pdf'. The cache computes a hash of
+    the file (and the version of the external software used to convert
+    the file) and if a result by that hash value is available, writes
+    the data to the output location supplied by the caller. Otherwise
+    the test case has to run the conversion and can then insert the
+    result into the cache.
+
+    Parameters
+    ----------
+    directory : str, optional
+        Files are stored in this directory, defaults to `'test_cache'` in
+        the overall Matplotlib cache directory.
+    max_size : int, optional
+        The flush method will delete files until their combined size is
+        under this limit, in bytes. Defaults to 100 megabytes.
+
+    """
+
+    def __init__(self, directory=None, max_size=int(1e8)):
+        self.gets = set()
```

Comment:
Now it's private so the names are easier to change later, but I wonder if you have a proposal for better names? I think "cache hit" is a common term and "get" has a nice symmetry with "hit".